subprojects/logging/src/main/java/org/gradle/internal/logging/slf4j/DefaultContextAwareTaskLogger.java

```diff
@@ -0,0 +1,481 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging.slf4j;
+
+import org.gradle.api.logging.LogLevel;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.internal.operations.CurrentBuildOperationRef;
+import org.gradle.internal.operations.OperationIdentifier;
+import org.slf4j.Marker;
+import org.slf4j.helpers.FormattingTuple;
+import org.slf4j.helpers.MessageFormatter;
+
+public class DefaultContextAwareTaskLogger implements ContextAwareTaskLogger {
+
+    private BuildOperationAwareLogger delegate;
+    private OperationIdentifier fallbackOperationIdentifier = null;
+
+    public DefaultContextAwareTaskLogger(Logger delegate) {
+        this.delegate = (BuildOperationAwareLogger) delegate;
+    }
+
+    public void setFallbackBuildOperationId(OperationIdentifier operationIdentifier) {
+        this.fallbackOperationIdentifier = operationIdentifier;
+    }
+
+    @Override
+    public boolean isLifecycleEnabled() {
+        return delegate.isLifecycleEnabled();
+    }
+
+    @Override
+    public boolean isQuietEnabled() {
+        return delegate.isQuietEnabled();
+    }
+
+    @Override
+    public boolean isEnabled(LogLevel level) {
+        return delegate.isEnabled(level);
+    }
+
+    @Override
+    public String getName() {
+        return delegate.getName();
+    }
+
+    @Override
+    public boolean isTraceEnabled() {
+        return delegate.isTraceEnabled();
+    }
+
+    @Override
+    public boolean isTraceEnabled(Marker marker) {
+        return delegate.isTraceEnabled(marker);
+    }
+
+    @Override
+    public boolean isDebugEnabled() {
+        return delegate.isDebugEnabled();
+    }
+
+    @Override
+    public boolean isDebugEnabled(Marker marker) {
+        return delegate.isDebugEnabled(marker);
+    }
+
+    @Override
+    public boolean isInfoEnabled() {
+        return delegate.isInfoEnabled();
+    }
+
+    @Override
+    public boolean isInfoEnabled(Marker marker) {
+        return delegate.isInfoEnabled(marker);
+    }
+
+    @Override
+    public boolean isWarnEnabled() {
+        return delegate.isWarnEnabled();
+    }
+
+    @Override
+    public boolean isWarnEnabled(Marker marker) {
+        return delegate.isWarnEnabled(marker);
+    }
+
+    @Override
+    public boolean isErrorEnabled() {
+        return delegate.isErrorEnabled();
+    }
+
+    @Override
+    public boolean isErrorEnabled(Marker marker) {
+        return delegate.isErrorEnabled(marker);
+    }
+
+    public void trace(String msg) {
```

Comment:
Do we just never do anything with trace logs? I don't really understand why these are treated differently to the other levels here.

Reply:
we just ignore trace level at all. gradle doesn't know anything bout trace